### PR TITLE
Add select method returning DataFrame

### PR DESCRIPTION
**Pull request by mikehinchey**
Add select method returning DataFrame

A convenience over execute, it collects the result cursor into a pandas DataFrame.

Also, params ipc and gpu_device let it choose ipc cpu or gpu results.

### CODE REVIEW

File: pymapd/connection.py [hunk 1]
```diff
@@ -295,6 +295,48 @@ def cursor(self):
...
+        Returns
+        -------
+        output: execution type dependent
+            If IPC and with no GPU: ``pandas.DataFrame``
+            If IPC and GPU: ``cudf.DataFrame``
+            If not IPC and with no GPU: ``pandas.DataFrame``
+        """
+        if ipc in (None, False) and gpu_device is None:
+            cursor = self.execute(operation)
+            return pd.DataFrame(
```
Comment (randyzwitch) [inline pymapd/connection.py]: This is the piece I was talking about in our call...you could replace this with `pd.read_sql("select * ...", conn)` and leave the logic to pandas. Or, not, it's not that big of a piece of code to maintain

**Comment by randyzwitch** [comment]
Retest this please.

**Comment by randyzwitch** [comment]
Jenkins retest this please

**Comment by randyzwitch** [comment]
Related: #246

If we're going to merge this, it feels like we should deprecate `select_ipc` and `select_ipc_gpu`, to avoid having the `load_table`, `load_table_columnar`, `load_table_arrow` confusion again

**Comment by jclay** [comment]
@mikehinchey OK to close this for now, or does it need to be revisited in the near-term?

**Comment by jclay** [comment]
Fixed in omnisci/pyomnisci

**Pull request closed without merging**